alchemiscale/storage/statestore.py

```diff
@@ -818,7 +816,8 @@ def get_transformation_failures(
         MATCH (trans:Transformation {_scoped_key: $scoped_key}),
               (trans)<-[:PERFORMS]-(:Task)-[:RESULTS_IN]->(res:ProtocolDAGResultRef)
         WHERE res.ok = false
-        RETURN res
+        WITH res._scoped_key as sk
+        RETURN DISTINCT sk
```

Comment:
Query but under what situations would it be possible to end up with duplicate sks here?

Reply:
We end up with them in the tests, where we push the same `ProtocolDAGResultRef` for two different `Task`s on the same `Transformation`. Not really a case we should see in real usage, but forcing `DISTINCT` here avoids the issue. We really do only want one instance of a given `ProtocolDAGResult` coming out of this method.